src/overlays/actors/ovl_Bg_Hidan_Rock/z_bg_hidan_rock.c

```diff
@@ -47,31 +61,336 @@ static ColliderCylinderInit D_8088BF8C = {
+                temp_v1 = Math_StepToF(&D_8088BFC0, -20.0f, this->dyna.actor.speedXZ);
+            }
+
+            this->dyna.actor.world.pos.x = (Math_SinS(this->dyna.unk_158) * D_8088BFC0) + this->dyna.actor.home.pos.x;
+            this->dyna.actor.world.pos.z = (Math_CosS(this->dyna.unk_158) * D_8088BFC0) + this->dyna.actor.home.pos.z;
+
+            if (temp_v1) {
+                player->stateFlags2 &= ~0x10;
+                this->dyna.unk_150 = 0.0f;
+                this->dyna.actor.home.pos.x = this->dyna.actor.world.pos.x;
+                this->dyna.actor.home.pos.z = this->dyna.actor.world.pos.z;
+                D_8088BFC0 = 0.0f;
+                this->dyna.actor.speedXZ = 0.0f;
+                this->timer = 5;
+            }
+
+            func_8002F974(&this->dyna.actor, NA_SE_EV_ROCK_SLIDE - SFX_FLAG);
+        } else {
+            player->stateFlags2 &= ~0x10;
+            this->dyna.unk_150 = 0.0f;
+            if (this->timer != 0) {
+                this->timer--;
+            }
+        }
+    } else {
+        this->timer = 0;
+    }
+
+    sp2C = this->dyna.actor.world.pos.z - D_8088BF60.z;
+    if (sp2C < 0.5f) {
+        Flags_SetSwitch(globalCtx, this->dyna.actor.params);
+        Math_Vec3f_Copy(&this->dyna.actor.home.pos, &D_8088BF60);
+        this->dyna.actor.world.pos.x = D_8088BF60.x;
+        this->dyna.actor.world.pos.z = D_8088BF60.z;
+        this->dyna.actor.speedXZ = 0.0f;
+        D_8088BFC0 = 0.0f;
+        player->stateFlags2 &= ~0x10;
+        this->actionFunc = func_8088B79C;
+    }
+
+    frame = globalCtx->gameplayFrames & 0xFF;
+    if (globalCtx->gameplayFrames & 0x100) {
+        this->unk_16C = 0.0f;
+    } else if (frame < 128) {
+        this->unk_16C = sinf(frame * (4 * 0.001f * M_PI)) * 19.625f;
+    } else if (frame < 230) {
+        this->unk_16C = 19.625f;
+    } else {
+        this->unk_16C -= 1.0f;
+        this->unk_16C = CLAMP_MIN(this->unk_16C, 0.0f);
+    }
+
+    if (sp2C < 100.0f) {
+        this->unk_16C = CLAMP_MAX(this->unk_16C, 6.125f);
+    }
+}
+
+void func_8088B5F4(BgHidanRock* this, GlobalContext* globalCtx) {
+    if (this->timer != 0) {
+        this->timer--;
+    }
+
+    if (this->timer == 0) {
+        func_8088B24C(this);
+    }
+}
+
+void func_8088B634(BgHidanRock* this, GlobalContext* globalCtx) {
+    if (func_8004356C(&this->dyna)) {
+        this->timer = 20;
+        this->dyna.actor.world.rot.y = Camera_GetCamDirYaw(ACTIVE_CAM) + 0x4000;
+        this->actionFunc = func_8088B69C;
+    }
+}
+
+void func_8088B69C(BgHidanRock* this, GlobalContext* globalCtx) {
+    if (this->timer != 0) {
+        this->timer--;
+    }
+
+    if (this->timer != 0) {
+        this->dyna.actor.world.pos.x =
+            this->dyna.actor.home.pos.x + 5.0f * Math_SinS((this->dyna.actor.world.rot.y + (this->timer << 14)));
+        this->dyna.actor.world.pos.z =
+            this->dyna.actor.home.pos.z + 5.0f * Math_CosS((this->dyna.actor.world.rot.y + (this->timer << 14)));
```

Comment:
```suggestion
        this->dyna.actor.world.pos.x =
            this->dyna.actor.home.pos.x + 5.0f * Math_SinS(this->dyna.actor.world.rot.y + (this->timer * 0x4000));
        this->dyna.actor.world.pos.z =
            this->dyna.actor.home.pos.z + 5.0f * Math_CosS(this->dyna.actor.world.rot.y + (this->timer * 0x4000));
```
This makes a little more sense since we're dealing with angles.